ds-collector/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,221 @@
+
+### Many Errors and Failures are "Okay"
+The Collector tries to get a lot of information from the environment, but not all commands and directory permissions allow the Collector to run completely without problem. The Collector is written in such a way that it survives these problems, failing only when a mandatory command is unable to complete successfully. 
+
+### Use `-d` Option While Troubleshooting
+Most problems are resolved by making adjustments to the configuration. Rather than getting errors from every node in the cluster, use the `-d` option to `ds-collector`; this will have the collector run on the `-n` node only, and it will not try to run on all the nodes in the cluster. It is by design that the Collector keeps going if one node has a problem; it is trying to get an understanding of the entire cluster and should not fail because one node has a problem.
+
+### Avoid Uploading Artifacts While Troubleshooting
+To avoid uploading of artifacts during testing, you can configure `skipS3="true"`.  Remember to comment this out (or set `false`) when you are ready to run the Collector "for real."
+
+### Customize the `update_env()` function
+In the `ds-collector` script is a customizable function:
+
+    update_env() {
+      # if nodes have specific environments that needs setting up/declaring, do so here
+      :
+    }
+
+This function can be edited to include any commands, and/or `export` any enviornment variables that need to be in place on each node before information is gathered. It is not a common requirement to do this - most environments are addressable through existing configuration, but just be aware that it is a customization point that can be used!
+
+### When Bastion Cannot Access AWS S3 to Upload
+Often, the Bastion exists on a network that cannot reach AWS. To work around this:
+- Configure `skipS3="true"`
+- Run `ds-collector` on the Bastion
+- Find a Unix environment that can access both the Bastion and AWS
+- Unpack `ds-collector` to this Unix environment, including the secret `.key` file
+- Transfer files from the Bastion base directory (e.g. `/tmp/datastax`) to this Unix environment; place them in the same base directory
+- Run `ds-collector` on this Linux environment, specifying the `-a` flag ("upload mode"); the collector will not run but the files will upload to S3. 
+
+### Files to be audited/obfuscated before uploading
+Some sites have requirements to audit and/or obfuscate the collected files before they are uploaded. To address this:
+- Configure `skipS3="true"`
+- Run `ds-collector`
+- When complete, files are in the base directory (e.g. `/tmp/datastax`) on the bastion as `.tar.gz` files. For each file in turn,
+    - Unpack the file
+    - Review/obfucate
+    - Repack into a `.tar.gz` file
+- Run `ds-collector`, specifying the `-a` flag ("upload mode"); the node collection process will will not run but the files will upload to S3
+
+
+## Known Problems and Issues
```

Comment:
as mentioned, as number of these items are already written up in the README.md

they have been written up with more detail here. i'm ok if all of the existing items are written up here and the existing Troubleshooting section simply refers to this file.